sycl/doc/extensions/proposed/sycl_ext_oneapi_device_image_backend_content.asciidoc

```diff
@@ -0,0 +1,255 @@
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.
+An implementation supporting this extension must predefine the macro
+`SYCL_EXT_ONEAPI_DEVICE_IMAGE_BACKEND_CONTENT` to one of the values defined in
+the table below.
+Applications can test for the existence of this macro to determine if the
+implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== New member functions in the `device_image` class
+
+This extension adds the following member functions to the `device_image` class.
+
+[source,c++]
+----
+namespace sycl {
+
+template <bundle_state State>
+class device_image {
+ public:
+  backend ext_oneapi_get_backend() const noexcept;
+  std::vector<std::byte> ext_oneapi_get_backend_content() const;
+
+#if defined(__cpp_lib_span)
+  std::span<std::byte> ext_oneapi_get_backend_content_view() const;
+#endif
+
+  /*...*/
+};
+
+} // namespace sycl
+----
+
+'''
+
+[frame=all,grid=none,separator="@"]
+!====
+a@
+[source,c++]
+----
+backend ext_oneapi_get_backend() const noexcept;
+----
+!====
+
+_Returns:_ The backend that is associated with this device image.
+This is always the same as the backend of the kernel bundle that contains this
+device image.
+
+'''
+
+[frame=all,grid=none,separator="@"]
+!====
+a@
+[source,c++]
+----
+std::vector<std::byte> ext_oneapi_get_backend_content() const;
+----
+!====
+
+_Constraints:_ Available only when `State` is `bundle_state::executable`.
+
+_Returns:_ A copy of the raw backend content for this device image.
+The format of this data is implementation-defined.
+See below for a description of the formats used by {dpcpp}.
+
+'''
+
+[frame=all,grid=none,separator="@"]
+!====
+a@
+[source,c++]
+----
+std::span<std::byte> ext_oneapi_get_content_backend_view() const;
+----
+!====
+
+Available only when the compiler defines the `__cpp_lib_span` feature-test macro
+(which is defined in {cpp}20 and higher).
+
+_Constraints:_ Available only when `State` is `bundle_state::executable`.
+
+_Returns:_ A view of the raw backend content for this device image.
+The data behind this view has the same lifetime as the `device_image` object.
+The format of this data is implementation-defined.
+See below for a description of the formats used by {dpcpp}.
+
+'''
+
+
+== Device image format for {dpcpp}
+
+This section is non-normative and applies only to the {dpcpp} implementation.
+The format of the data returned by
+`device_image::ext_oneapi_get_backend_content` and
+`device_image::ext_oneapi_get_backend_content_view` depends on the backend of the
+kernel bundle that contains the device image.
+
+=== Format on Level Zero
+
+The device image's backend content is native ISA for the device, which can be
```

Comment:
How can we guarantee that? I don't see any restrictions in the core SYCL 2020 specification preventing iteration over device images of an `input` kernel bundle. Is it expected that `ext_oneapi_get_backend_content` may invoke JIT compilation if used on `input` or `object` kernel bundle using Level Zero backend?

Reply:
Apology, missed the constraint on the API